src/mlpack/methods/ann/layer/weight_norm.hpp

```diff
@@ -137,6 +137,9 @@ class WeightNorm
   //! Get the wrapped layer.
   LayerTypes<CustomLayers...> const& Layer() { return wrappedLayer; }
 
+  //! Get size of weights.
+  size_t WeightSize() const { return layerWeightSize + 1; }
```

Comment:
This code assumes that `layerWeightSize` is set in a valid way, but that's only done in the constructor.  It might be "safer" to remove `layerWeightSize` as a variable from the class, and then simply use a `WeightSizeVisitor` in this `WeightSize()` call.  That way, if the size of whatever layer is being held changes, then `WeightSize()` here will also reflect that change. :+1: